app.rb

```diff
@@ -0,0 +1,119 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    i = 0
+    new_array = self
+    new_array = to_a if is_a?(Range) || is_a?(Hash)
+    while i < new_array.length
+      yield new_array[i]
+      i += 1
+    end
+    new_array
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    i = 0
+    new_array = self
+    new_array = to_a if is_a?(Range) || is_a?(Hash)
+    while i < new_array.length
+      yield new_array[i], i
+      i += 1
+    end
+    new_array
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    new_numbers = []
+    my_each { |i| new_numbers.push(i) if yield(i) }
+    new_numbers
+  end
+
+  def my_all?(arg = nil)
+    return false if arg.nil? && !block_given? && !my_each { |j| j }
+
+    unless arg.nil?
+      return false unless my_each { |j| j.class == arg || arg.match(j) }
+    end
+    return false if block_given? && !my_each { |j| yield(j) }
+
+    true
+  end
+
+  def my_any?(arg = nil)
+    return true unless arg.nil? || !my_each { |j| j.class == arg }
+    return true unless block_given? || !my_each { |j| j }
+    return true if block_given? && my_each { |j| yield(j) }
+
+    true
+  end
```

Comment:
# my_any?
1. [ ] Returns false if the block ever returns a value false or nil
## Test 👉 `array.my_any?(&false_block)`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
#
2. [ ] When no block or argument is given returns false if at least one of the collection is not true
### Test 👉 `false_array.my_any?`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
#
3. [ ] When a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer
### Test 👉 `words.my_any?(Integer)`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
#
4. [ ]  When a Regex is passed as an argument returns false if none of the collection matches the Regex
### Test 👉 `words.my_any?(/z/)`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
#
5. [ ] When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
### Test 👉 `words.my_any?('cat')`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
#